hyperspy/gui/peakfinder2D.py

```diff
@@ -0,0 +1,288 @@
+import inspect
+from collections import OrderedDict
+import numpy as np
+import matplotlib.pyplot as plt
+from IPython.display import display
+from hyperspy._signals.signal2d import find_peaks_max, find_peaks_minmax, \
+    find_peaks_zaefferer, find_peaks_stat, find_peaks_dog, find_peaks_log
+
+METHODS = [find_peaks_max, find_peaks_minmax, find_peaks_zaefferer,
+           find_peaks_stat, find_peaks_dog, find_peaks_log]
+
+
+class PeakFinderUIBase:
+
+    def __init__(self):
+        self.signal = None
+        self.indices = None
+        self.methods = METHODS
+        self.method_names = [method.__name__ for method in METHODS]
+        self.params = {method.__name__: OrderedDict(
+            [(p.name, p.default) for p in
+             inspect.signature(method).parameters.values() if
+             p.default is not inspect._empty]) for method in METHODS}
+        self._method = self.method_names[0]
+
+    def interactive(self, signal):
+        self.signal = signal
+        self.indices = self.signal.axes_manager.indices
+        self.init_ui()
+
+    def init_ui(self):
+        raise NotImplementedError
+
+    @property
+    def current_method(self):
+        return dict(zip(self.method_names, self.methods))[self._method]
+
+    def get_data(self):
+        return self.signal.inav[self.indices].data
```

Comment:
This also recreates the full signal, copying many things that are not necessary in the end - I suggest replacing it by the following (feel free to combine this to a one-liner, just splitting for clarity):
```python
_slices = self.signal._get_array_slices(self.indices, isNavigation=True)
return self.signal.data[_slices]
```